src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -1,195 +1,195 @@
+                final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber);
+                filesToDelete.put(bucketName, filesProcessed);
             }
-            logger.log("waiting for file to be created " + summaryUpdateName);
-            Thread.sleep(1000);
-        }
-        
-        // Before we delete the shipment updates make sure the summary update file exists
-        if (EventHandler.getS3Client().doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
-            deleteProcessedFiles(filesToDelete);
-            logger.log("All updates successfully processed");
-        } else {
-            throw new RuntimeException("Failed to write summary status, will be retried in 15 minutes");
+            final AmazonS3 s3Client = EventHandler.getS3Client();
+
+            //Create a new file in the Constants.SUMMARY_BUCKET
+            logger.log("Map of statuses -> " + latestStatusForTrackingNumber);
+            String summaryUpdateName = Long.toString(System.currentTimeMillis());
+
+            EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+
+            long expirationTime = System.currentTimeMillis() + Duration.ofMinutes(1).toMillis();
+            while(System.currentTimeMillis() < expirationTime) {
+                if (s3Client.doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
+                    break;
+                }
+                logger.log("waiting for file to be created " + summaryUpdateName);
+                Thread.sleep(1000);
+            }
+
+            // Before we delete the shipment updates make sure the summary update file exists
+            if (EventHandler.getS3Client().doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
+                deleteProcessedFiles(filesToDelete);
+                logger.log("All updates successfully processed");
+            } else {
+                throw new RuntimeException("Failed to write summary status, will be retried in 15 minutes");
+            }
+
         }
-        
-    }
 
-    private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, Map<String, Pair<Long, String>> latestStatusForTrackingNumber) {
-        final AmazonS3 s3Client = EventHandler.getS3Client();
-        logger.log("Processing Bucket: " + bucketName);
+        private List<KeyVersion> processEventsInBucket(String bucketName, LambdaLogger logger, Map<String, Pair<Long, String>> latestStatusForTrackingNumber) {
+            final AmazonS3 s3Client = EventHandler.getS3Client();
+            logger.log("Processing Bucket: " + bucketName);
 
-        ObjectListing files = s3Client.listObjects(bucketName);
-        List<KeyVersion> filesProcessed = new ArrayList<DeleteObjectsRequest.KeyVersion>();
+            ObjectListing files = s3Client.listObjects(bucketName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.